blockchain/api.go

```diff
@@ -174,6 +182,46 @@ func (bch *api) OpenDeal(key *ecdsa.PrivateKey, deal *pb.Deal) (*types.Transacti
 	return tx, err
 }
 
+func (bch *api) OpenDealPending(key *ecdsa.PrivateKey, deal *pb.Deal, wait time.Duration) (*big.Int, error) {
+	tx, err := bch.OpenDeal(key, deal)
+	if err != nil {
+		return nil, err
+	}
+
+	fmt.Println(tx.Hash().String())
+
+	ctx, cancel := context.WithTimeout(context.Background(), wait)
+	defer cancel()
+
+	tk := time.NewTicker(1 * time.Second)
+	defer tk.Stop()
+
+	for {
+		select {
+		case <-tk.C:
+			txReceipt, err := bch.client.TransactionReceipt(ctx, tx.Hash())
+			if err != nil {
+				if err == ethereum.NotFound {
+					break
+				}
+				return nil, err
+			}
+			if txReceipt.Status == types.ReceiptStatusSuccessful {
+				for _, l := range txReceipt.Logs {
+					if l.Topics[0] == DealOpenedTopic {
+						return l.Topics[3].Big(), nil
```

Comment:
Are those indices guaranteed to exist? I have no idea, just asking.

Reply:
slice indices without length checking...? Looks fragile.